readthedocs/projects/tasks.py

```diff
@@ -670,6 +670,8 @@ def run_setup(self, record=True):
 
         Return True if successful.
         """
+        api_v2.build(self.build['id']).restart.post()
```

Comment:
I'd like to migrate soon to #3984 to put this code on the proper methods (eg. `on_failure`) instead of in random places.